src/commands/kv/bucket/mod.rs

```diff
@@ -113,7 +113,11 @@ pub fn generate_url_safe_key_and_hash(
 
     let path_with_hash = if let Some(value) = value {
         let digest = get_digest(value)?;
-        format!("{}-{}", url_safe_path, digest)
+
+        generate_path_with_hash(path, digest)?
```

Comment:
this `unwrap` makes me uneasy

Reply:
We should just be able to use `?` instead, correct?

Reply:
pushed an update based on in air convo w/ @EverlastingBugstopper